### ISSUE TITLE
Additional bio-geochemical sequences for data from Argo profiling floats 

### ISSUE THREAD

**Issue by efucile** [wmo-im/BUFR4#5]
Add entries:

in BUFR/CREX Table B,
Class 13 – Hydrographic and hydrological elements
Table reference	Element name	BUFR	CREX
F	XX	YYY		Unit	Scale	Ref. value	Data width (bits)	Units	Scale	Data width (characters)
0	13	161	pH scale	Code table	0	0	3	Code table	0	1

Class 41 – Marine bio-geochemical data
Table reference	Element name	BUFR	CREX
F	XX	YYY		Unit	Scale	Ref. value	Data width (bits)	Units	Scale	Data width (characters)
0	41	006	Backscattering	m-1	5	0	19	m-1	5	6

in BUFR Table D,
Sequence 3-06-044 for chlorophyll-A profile data

Table Reference	Table References	Element Name	
F	X	Y			
3	06	044				(Chlorophyll-A (fluorescence) profile data)	
	1	09	000	Delayed replication of 9 descriptors	
	0	31	002	Extended delayed descriptor replication factor	Gives number of depths
	0	07	062	Depth below sea / water surface	Code as missing
	0	08	080	Qualifier for quality class 	Code as missing
	0	33	050	GTSPP quality class	Code as missing
	0	07	065	Water pressure	
	0	08	080	Qualifier for quality class (set to 10, indicates pressure at a level)	
	0	33	050	GTSPP quality class	
	0	41	002	Chlorophyll-A (fluorescence)	In kg l-1 (= 109 mg m-3)
	0	08	080	Qualifier for quality class (set to 21, chlorophyll-A at a level)	
	0	33	050	GTSPP quality class	

Code depth related descriptors as missing as water pressure is used as the vertical axis
Chlorophyll-A specified in range 0 to 65.535 mg m-3 with a resolution of 0.001 mg m-3.

Sequence 3-06-045 for dissolved nitrate profile data

Table Reference	Table References	Element Name	
F	X	Y			
3	06	045				(Dissolved nitrate profile data)	
	1	09	000	Delayed replication of 9 descriptors	
	0	31	002	Extended delayed descriptor replication factor	Gives number of depths
	0	07	062	Depth below sea / water surface	Code as missing
	0	08	080	Qualifier for quality class 	Code as missing
	0	33	050	GTSPP quality class	Code as missing
	0	07	065	Water pressure	
	0	08	080	Qualifier for quality class (set to 10, indicates pressure at a level)	
	0	33	050	GTSPP quality class	
	0	41	003	Dissolved nitrate	In μmol kg-1
	0	08	080	Qualifier for quality class (set to 22, nitrate at a level)	
	0	33	050	GTSPP quality class	

Code depth related descriptors as missing as water pressure is used as the vertical axis.

Sequence 3-06-046 for pH profile data

Table Reference	Table References	Element Name	
F	X	Y			
3	06	046				(pH profile data)	
	1	09	000	Delayed replication of 9 descriptors	
	0	31	002	Extended delayed descriptor replication factor	Gives number of depths
	0	07	062	Depth below sea / water surface	Code as missing
	0	08	080	Qualifier for quality class 	Code as missing
	0	33	050	GTSPP quality class	Code as missing
	0	07	065	Water pressure	
	0	08	080	Qualifier for quality class (set to 10, indicates pressure at a level)	
	0	33	050	GTSPP quality class	
	0    	13	161	pH scale	
	0	13	080	pH	dimensionless
	0	08	080	Qualifier for quality class (set to 23, pH at a level)	
	0	33	050	GTSPP quality class	

Code depth related descriptors as missing as water pressure is used as the vertical axis.

Sequence 3-06-047 for backscattering profile data
Table Reference	Table References	Element Name	
F	X	Y			
3	06	047				(Backscatter profile data)	
			0	02	071	(Spectrographic) wavelength	m
	1	09	000	Delayed replication of 9 descriptors	
	0	31	002	Extended delayed descriptor replication factor	Gives number of depths
	0	07	062	Depth below sea / water surface	Code as missing
	0	08	080	Qualifier for quality class 	Code as missing
	0	33	050	GTSPP quality class	Code as missing
	0	07	065	Water pressure	
	0	08	080	Qualifier for quality class (set to 10, indicates pressure at a level)	
	0	33	050	GTSPP quality class	
	0	41	006	Backscattering	m-1
	0	08	080	Qualifier for quality class (set to 24, backscatter at a level)	
	0	33	050	GTSPP quality class	
Code depth related descriptors as missing as water pressure is used as the vertical axis.

in BUFR/CREX code tables,
0 02 149
Type of data buoy

Code figure	Meaning
31	Coastal sub-surface float
32	Deep sub-surface float

0 08 080
Qualifier for GTSPP Quality Flag

Code figure	Meaning
21	Chlorophyll-A at a level
22	Nitrate at a level
23	pH at a level
24	Backscattering at a level

0 22 067 (FT2017-2)
Instrument type for water temperature/salinity profile measurement

Code figure	Meaning
835	PROVOR IV
836	PROVOR III
870	HM2000
871	COPEX
872	S2X
869	DOVA (no longer required)
870	NAMI (no longer required)
871	HM2000

Add a code table:
0 13 161
pH scale

Code figure	Meaning
0	Seawater scale
1	Freescale
2	Total scale
3-6	Reserved
7	missing

**Comment by DavidBerryNOC** [comment]
Validation still waiting input / testing by Argo team.

**Comment by DavidBerryNOC** [comment]
I've moved to FT-2021-1 but doubt it will be done by then.

**Comment by amilan17** [comment]
This issue needs a branch.

**Comment by amilan17** [comment]
https://github.com/wmo-im/BUFR4/tree/issue-5

**Comment by amilan17** [comment]
Who is responsible for validation?

**Comment by marijanacrepulja** [comment]
The branch is not updated with proposed BUFR entries.
I am happy to help with validation once the branch is ready.

**Comment by DavidBerryNOC** [comment]
The last time I checked (September / October of this year) there was still no progress to report. There may still be some changes to make so I would propose moving this back to in progress rather than in validation and changing the FT aimed for.

**Comment by amilan17** [comment]
@jitsukoh @DavidBerryNOC 

Change target FT to 2021-2?

**Comment by jitsukoh** [comment]
@amilan17 I would prefer to set target FT when we know the validation process really starts. So if there is no update, I'd suggest not to add a target FT right now. It would be good if we can document this process (when and who add the target FT) to the Github status management document...

**Comment by amilan17** [comment]
re: [https://github.com/wmo-im/CCT/wiki/Teleconference-26-and-28.4.2022](https://github.com/wmo-im/CCT/wiki/Teleconference-26-and-28.4.2022)

<table><tbody><tr><td>Delayed by issue <span class="mention" data-mention="#16">#16</span> &nbsp;(Glider template).</td></tr></tbody></table>

###

**Comment by amilan17** [comment]
@JonTurton, Can you please open a new issue with the updated proposal when it's ready? That new issue will then supersede this one. If you do this by March timeframe, it can be included in the FT2023-2 proposals that will be developed late spring/early summer of 2023.  Thanks!

**Comment by JonTurton** [comment]
@amilan17 , Hi Anna- I've just contacted colleagues in the ADMT to check on progress and when they anticipate providing us with the information required (units, range and precision) for each BGC variable. Once we have that putting together the sequences should be relatively straightforward if we follow that already used for dissolved oxygen (as its basically just changing the descriptor for the variable).  Once we have that, I can open as a new issue, but we may not have all the info needed by March.

**Comment by JonTurton** [comment]
Hi Anna,

I have now entered this as a new issue in the GitHub (https://github.com/wmo-im/BUFR4/issues/155).  Hopefully it all looks OK (I've not used it before).

Jon
